src/components/alert/alert.js

```diff
@@ -63,9 +63,10 @@ export default {
     dismiss() {
       this.clearCounter()
       if (typeof this.show === 'number') {
-        this.$emit('dismiss-count-down', 0)
```

Comment:
If someone clicks the `dismiss` close button, the countdown timer value will not be emitted as 0 (and some users rely on this event emitting 0 when dismissed early.

Reply:
you should check first if `this.dismissCountDown` is greater than 0, and if so, emit `'dismiss-count-down' with a value of `0`

Reply:
That's the original code - I've just added an extra event to emit `dismissed`.

I do agree though, but there is no `this.dismissCountDown` - it's a local variable in the `showChanges` method.

I can add another `parseInt`  on `this.show` and then if it's greater than 0, emit the event?

Reply:
Ah right... in the revamped code I was working on for BAlert, I was switching to `setTimeout` from `setInterval` (to simplify the code/logic) and storing the `dismissCountDown` value  in `data()`.

Maybe at line 95 where we defined the variable `dismissCountDown`, it should be stored in data, and any decrement would be done on the `this.dismissCountDown`.

Reply:
Or, I can update the BAlert code with the new code I was getting ready to implement and see if it fixes the original issue (as it also fixes a few other issues with BAlert)

Reply:
@tmorehouse I'm not fussy - I just need to see this fixed as I'm planning on using it heavily.